src/phases/egg-hatch-phase.ts

```diff
@@ -98,6 +98,8 @@ export class EggHatchPhase extends Phase {
 
       globalScene.audioManager.fadeOutBgm(undefined, false);
 
+      // TODO: the hatch phase and ui handler should not be intertwined in this way;
+      // the phase also should not be the one creating the graphical objects
       this.eggHatchHandler = globalScene.ui.getHandler() as EggHatchSceneUiHandler;
```

Comment:
NIT: Possibly out of scope
```suggestion
      this.eggHatchHandler = globalScene.ui.getHandler<EggHatchSceneUiHandler;>();
```